src/Simplex/Chat.hs

```diff
@@ -1655,6 +1655,43 @@ processChatCommand' vr = \case
     case conn'_ of
       Just conn' -> pure $ CRConnectionIncognitoUpdated user conn'
       Nothing -> throwChatError CEConnectionIncognitoChangeProhibited
+  APISetConnectionUserId connId newUserId -> withUser $ \user@User {userId} -> do
+    cfg <- asks config
+    newUser <- privateGetUser newUserId
+    conn' <- withFastStore $ \db -> do
+      conn <- getPendingContactConnection db userId connId
+      newUserServers <- liftIO $ getServers db cfg newUser SPSMP
+      pure $ checkConnection conn newUserServers
+    case conn' of
+      (True, Just conn) -> do
+        -- Safe to update connection
+        withAgent $ \a -> changeConnectionUser a (aUserId user) (aConnId' conn) (aUserId newUser)
+        _ <- withFastStore' $ \db -> updatePCCUser db userId conn newUserId
+        pure $ CRConnectionUserChanged user conn newUser
+      (False, Just conn) -> do
+        -- Servers do not match, create new connection
+        subMode <- chatReadVar subscriptionMode
+        (newConnId, cReq) <- withAgent $ \a -> createConnection a (aUserId user) True SCMInvitation Nothing IKPQOn subMode
+        newConn <- withFastStore' $ \db -> do
+          deleteConnectionRecord db user connId
+          createDirectConnection db newUser newConnId cReq ConnNew Nothing subMode initialChatVersion PQSupportOn
+        withAgent $ \a -> deleteConnectionAsync a False (aConnId' conn)
```

Comment:
I see this is still used in a couple of places (not sure why - maybe there was reason, maybe got overlooked), but I think here we should just use `deleteAgentConnectionAsync` to catch as we already replaced connection in db anyway so there's no recovery for failing to delete it